java-frontend/src/main/java/org/sonar/java/Measurer.java

```diff
@@ -111,12 +112,20 @@ public void scanFile(JavaFileScannerContext context) {
 
     RangeDistributionBuilder fileComplexityDistribution = new RangeDistributionBuilder(LIMITS_COMPLEXITY_FILES);
     saveMetricOnFile(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, fileComplexityDistribution.add(fileComplexity).build());
+
+    if (isGreaterThanOrEqualToSonarQube6_3()) {
+      saveMetricOnFile(CoreMetrics.COGNITIVE_COMPLEXITY, context.compilationUnitCognitiveComplexity());
+    }
   }
 
   private boolean isSonarLintContext() {
     return sensorContext.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(6, 0)) && sensorContext.runtime().getProduct() == SonarProduct.SONARLINT;
   }
 
+  private boolean isGreaterThanOrEqualToSonarQube6_3() {
```

Comment:
this is written `isSQGreaterThan62` in sonarComponents, I would use the same naming.